Automattic/jetpack
# Beta plugin: Support managing multiple plugins

#### Changes proposed in this Pull Request:

We want to be able to support testing plugins other than Jetpack, such as the upcoming Jetpack Backup plugin. We've already changed the build, and the change to the Beta Download website is pending. This updates the plugin itself.

#### Jetpack product discussion


p9dueE-362-p2

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






* Create a JN site, with Jetpack and Jetpack Beta but not VaultPress.
* SSH in to the JN site, and check this out:
  * `git clone --branch add/beta-plugin-multiple-plugins --depth=20 https://github.com/Automattic/jetpack`
  * `cd jetpack/projects/plugins/beta/`
  * `composer install`
  * `cd ~/apps/$USER/public/wp-content/plugins`
  * `rm -rf jetpack-beta`
  * `ln -s ~/jetpack/projects/plugins/beta jetpack-beta`
* Try using it to manage the plugins.
  * Switch between various versions of Jetpack.
  * Install "Bleeding Edge" VaultPress.
  * Go to the plugins page and disable VaultPress. It should completely disappear, since no "stable" version has been installed.
  * Install an old version of Jetpack. See that it shows that updates are available in JP Beta's management pages. See that the "Update" buttons there work.
  * Install a PR branch for a plugin, then update the PR in GitHub. See that it shows that updates are available in JP Beta's management pages. See that the "Update" buttons there work.
* Create another JN site, this time with multisite enabled. Set it up with JPBeta from this PR as above.
  * Enable Jetpack at the site level. Things should work exactly as above.
  * Enable Jetpack sitewide. All links to the JPBeta management page for Jetpack should now point to Network Admin rather than regular admin. Attempting to go to the regular admin URL should redirect to Network Admin.
